tools/grab_spotify_saved_tracks.py

```
@@ -0,0 +1,175 @@
+        
+        # After authentication, save the token back to environment if needed
+        if os.path.exists(cache_path):
+            with open(cache_path, 'r') as f:
+                token_info = f.read()
+                # In a GitHub Action, you might want to set this as an output
+                # to save for the next run
+                print(f"::set-output name=spotify_token::{token_info}")
+        
+        return sp
+    except Exception as e:
+        logging.error(f"Failed to initialize Spotify client: {e}")
```

Ruff (0.8.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)